packages/lime-system/files/usr/lib/lua/lime/utils.lua

```diff
@@ -296,4 +297,35 @@ function utils.set_password(username, password)
 									pass, pass, user))
 end
 
+function utils.get_root_secret()
+	local f = io.open(utils.SHADOW_FILENAME, "r")
+	if f ~= nil then
+		local root_line = f:read("*l") --! root user is always in the first line
```

Comment:
And if it is not in the first line?
Seems an unnecessary simplification

Reply:
From https://linuxize.com/post/etc-shadow-file/

> Typically, the first line describes the root user, followed by the system and normal user accounts.

so it is not forced to be in the first line.

Reply:
Thanks for the review!!!
Yes!  Albeit it is not forced, that is the way it is generated in OpenWrt (and in most linux systems, as this is part of the "first boot/bootstrap" of the system). Some user may change the order manually but it is very rare to edit this file by hand.
At the time of writing these lines of code, I have chosen to write a way simpler but not always correct code (specially the `set_root_secret`). I am open to write the longer but correct version. Thoughts?

Reply:
ok with me :)